sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_quickpulse/_live_metrics.py

```diff
@@ -195,13 +197,14 @@ def _record_span(self, span: ReadableSpan) -> None:
                         self._dependency_failure_rate_counter.add(1)
                     self._dependency_duration.record(duration_ms)
 
-                metric_infos_dict = _get_quickpulse_derived_metric_infos()
-                # check if filtering is enabled
-                if metric_infos_dict:
-                    # Derive metrics for quickpulse filtering
-                    data = _TelemetryData._from_span(span)
-                    _derive_metrics_from_telemetry_data(data)
-                    # TODO: derive exception metrics from span events
+                # Derive metrics for quickpulse filtering
+                data = _TelemetryData._from_span(span)
+                _derive_metrics_from_telemetry_data(data)
+
+                # Process docs for quickpulse filtering
+                _apply_document_filters_from_telemetry_data(data)
+
+                # TODO: derive exception metrics from span events
```

Comment:
There's a good amount happening in the rewriting of this file. Might need a walkthrough.

Reply:
_apply_document_filters_from_telemetry_data conditionally appends doc